pkg/provider/handlers/utils.go

```diff
@@ -0,0 +1,25 @@
+package handlers
+
+import (
+	"net/http"
+	"path"
+
+	faasd "github.com/openfaas/faasd/pkg"
+)
+
+func getRequestNamespace(namespace string) string {
+
+	if len(namespace) > 0 {
+		return namespace
+	}
+	return faasd.FunctionNamespace
+}
+
+func readNamespaceFromQuery(r *http.Request) string {
+	q := r.URL.Query()
+	return q.Get("namespace")
+}
+
+func getNamespaceSecretMountPath(userSecretPath string, namespace string) string {
```

Comment:
Consider the example of creating a secret on a previous version:

`faas-cli secret create test --from-literal api-token`

We will need a migration task on start-up to move the secrets from

/var/lib/faasd-provider/secrets/test

to

/var/lib/faasd-provider/secrets/openfaas-fn/test

Reply:
Done.

Reply:
@alexellis

Reply:
Great, thank you for confirming.